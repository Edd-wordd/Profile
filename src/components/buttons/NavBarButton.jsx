import React from 'react'
import { Button } from '@mui/material'

function NavButton({ tab, theme, windowLocation }) {
  return (
    <Button
      key={tab.route}
      href={tab.route}
      sx={{
        fontSize: 16,
        fontWeight: 500,
        padding: '1.5rem 0 1.5rem 0',
        color: theme.palette.defaultLight.main,
        letterSpacing: '.3rem',
        transitionTimingFunction: 'ease-in',
        transition: '.5s',
        '&:hover': {
          color: theme.palette.primary_300.main,
          backgroundColor: 'transparent',
          textShadow: `1px 1px ${theme.palette.primary_300.main}`,
        },
        ...(windowLocation.pathname === tab.route && {
          color: theme.palette.primary_300.main,
          textShadow: `1px 1px ${theme.palette.primary_300.main}`,
          marginBottom: '.5rem',
          textDecoration: 'line-through',
          '&:hover': {
            textDecoration: 'line-through',
          },
        }),
      }}
    >
      {tab.label}
    </Button>
  )
}

export default NavButton
