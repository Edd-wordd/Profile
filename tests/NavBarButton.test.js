import React from 'react'
import testComponent from './testComponent'
import NavBarButton from '../src/components/buttons/NavBarButton'

testComponent(<NavBarButton />, 'NavBarButton')
