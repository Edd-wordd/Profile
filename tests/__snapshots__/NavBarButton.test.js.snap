// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarButton should render NavBarButton 1`] = `"Component"`;
